### ISSUE TITLE
setWidth(),setHeight(), setSp() are 0 in release mode. 

### ISSUE THREAD

**Issue by ganger1** [OpenFlutter/flutter_screenutil#141]
Hi, When I force close my App from the android Recent Activity by sliding than open App on click Notification all methods giving 0 values. Text and Widgets height becomes 0. I also use V2.3.1 and 3.0.0 beta2 . I am creating my Custom theme.

[✓] Flutter (Channel unknown, 1.20.0-7.4.pre, on Mac OS X 10.15.2 19C57, locale en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
[✓] Android Studio (version 4.0)
[✓] Connected device (1 available)

• No issues found!

    MaterialApp(
                  key: _mKey,
                  navigatorObservers: [
                    sl<FirebaseAnalyticsService>().getObserver(),
                  ],
                  builder: (context,_){
                    return Scaffold(
                      key: _scaffoldkey,
                      body: Container(
                        color: Colors.white,
                        child: ExtendedNavigator<Router>(
                          router: Router(),
                          initialRoute: Routes.screenNavigatorHandler,
                        ),
                      ),
                    );
                  },
                  locale: _locale,
                  supportedLocales: [
                    Locale('en','GB'),//english 
                    Locale('ar','SA')// arabic
                  ],
                  localizationsDelegates: [
                    Localization.delegate,
                    GlobalMaterialLocalizations.delegate,
                    GlobalWidgetsLocalizations.delegate,
                    GlobalCupertinoLocalizations.delegate
                  ],
                  localeResolutionCallback: (Locale locale, Iterable<Locale> supportedLocales) {
                    if (locale == null) {
                      return supportedLocales.first;
                    }

                    for (Locale supportedLocale in supportedLocales) {
                      if (supportedLocale.languageCode == locale.languageCode ||
                          supportedLocale.countryCode == locale.countryCode) {
                        return supportedLocale;
                      }
                    }
                    return supportedLocales.first;
                  },
                  debugShowCheckedModeBanner: false,
                  title: 'App',
                  theme: ThemeData(
                      cupertinoOverrideTheme: CupertinoThemeData(
                        primaryColor: colorPrimary,
                      ),
                      accentColor: Colors.white,
                      appBarTheme:
                      AppBarTheme(textTheme: TextTheme(title: appBarTextStyle)),
                      primarySwatch: colorWhite,
                      dialogTheme: DialogTheme(
                        titleTextStyle: textStyleDisplay2,
                        contentTextStyle: body1TextStyle,
                      ),
                      textTheme: TextTheme(
                          title: appBarTextStyle,
                          body1: body1TextStyle,
                          body2: body2TextStyle,
                          button: buttonTextStyle,
                          subtitle: subTitleStyle,
                          display3: textStyleDisplay3,
                          display2: textStyleDisplay2,
                          display1: textStyleDisplay1,
                          caption: categoryTextStyle,
                          subhead: subHeadingStyle),
                      inputDecorationTheme: InputDecorationTheme(
                          border: inputFocusedBorder,
                          isDense: true,
                          alignLabelWithHint: true,
                          floatingLabelBehavior: FloatingLabelBehavior.never,
                          focusedBorder: inputFocusedBorder,
                          enabledBorder: inputEnabledBorder,
                          errorBorder: errorFocusedBorder,
                          focusedErrorBorder: errorFocusedBorder,
                          errorStyle: errorLabelStyle,
                          labelStyle: inputLabelTextStyle
                      )
                  ),
                  onGenerateRoute: Router().onGenerateRoute,
                  initialRoute: Routes.screenNavigatorHandler,
                )

`void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await di.init();
  _setupLogging();
  initializeFCM();
  ScreenUtil.init(designSize: Size(375,812), allowFontScaling: true);
  runApp(MainApp());
}`

**Comment by lizhuoyuan** [comment]
I didn't see the initialization code, where did you initialize it?
Click the notification to enter the native page and then enter the flutter page?

**Comment by ganger1** [comment]
here i initialize ScreenUtils
`    void main() async { WidgetsFlutterBinding.ensureInitialized(); await di.init(); _setupLogging(); initializeFCM(); ScreenUtil.init(designSize: Size(375,812), allowFontScaling: true); runApp(MainApp()); }`

**Comment by lizhuoyuan** [comment]
If it is a mixed project, if you enter the Flutter page from Android Activity, if the entered flutter page is not entered through main dart, it needs to be initialized separately.

like this:
SplashActivity -> FirstFluuterPage.
You can try to initialize in FirstFluuterPage's build()

**Comment by ganger1** [comment]
Yeah i tried to initialize in initState() method also tried in build() method of first page. But facing same issue.

**Comment by lizhuoyuan** [comment]
Maybe you need to debug, you can see if you can get the screen data through the official api during initialization?

**Comment by niugengyun** [comment]
作者出来唠两句啊 ,相同的问题，你的代码直接在下面的环境跑一下就可以复现：

`
[✓] Flutter (Channel master, 1.21.0-6.0.pre.142, on Mac OS X 10.15.6 19G73, locale en-CN)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.47.3)
[✓] Connected device (1 available)
`
